packages/server/src/unstable-core-do-not-import/stream/sse.ts

```diff
@@ -147,9 +154,19 @@ export function sseStreamProducer(opts: {
       }
 
       stream.controller.enqueue(chunk);
+
+      if (opts.emitAndEndImmediately) {
+        // end the stream in the next tick so that we can send a few more events from the queue
+        setTimeout(maxDurationPromise.resolve, 1);
```

Comment:
I'm not sure I understand the `1` here. Would this be a use-case for `setImmediate`? Or just `setTimeout(..., 0)`?